Clean up stale issue links

stephentoub opened the pull request:
- In some cases, I deleted ActiveIssues associated with issues where the problem was supposedly fixed.  We'll see.
- In some cases, I deleted test cases entirely, e.g. where the corresponding issue said effectively "we'll live with the behavior" or "another repo is responsible for this now" (paraphrasing), or where the tests were initially added to the repo years ago already disabled and against bug numbers from some old bug database.
- In some cases, I replaced the link with the correct one.
- In some cases, I re-opened issues.

stephentoub commented:
For reference and for anyone else who might want to play around with finding stale issues I may have missed, I ran variations on this:
```C#
using System;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using Octokit;

class Program
{
    static async Task Main()
    {
        const string Token = "REPLACETHIS"; // get token from GitHub personal tokens settings page
        const string Repo = "corefx";
        const string Path = @"d:\repos\runtime\src\libraries\";

        var client = new GitHubClient(new ProductHeaderValue("ClosedIssueSearcher")) { Credentials = new Credentials(Token) };
        var issueRegex = new Regex($"https://github.com/dotnet/{Repo}/issues/(\\d+)", RegexOptions.Compiled);
        var results = (from path in Directory.EnumerateFiles(Path, "*.cs", SearchOption.AllDirectories)
                       from line in File.ReadLines(path)
                       where line.Contains("TODO", StringComparison.OrdinalIgnoreCase) || line.Contains("ActiveIssue")
                       let matches = issueRegex.Matches(line)
                       where matches.Count > 0
                       from match in matches
                       select int.Parse(match.Groups[1].Value)).Distinct();

        foreach (int id in results)
        {
            try
            {
                Issue issue = await client.Issue.Get("dotnet", Repo, id);
                if (issue.State.StringValue == "closed")
                {
                    Console.WriteLine($"{issue.State.StringValue}: https://github.com/dotnet/{Repo}/issues/{id}");
                }
            }
            catch (NotFoundException)
            {
                Console.WriteLine($"Not found: https://github.com/dotnet/{Repo}/issues/{id}");
            }
        }
    }
}
```

danmoseley reviewed (approved):
thanks for doing this

stephentoub committed:
Clean up stale issue links

- In some cases, I deleted ActiveIssues associated with issues where the problem was supposedly fixed.  We'll see.
- In some cases, I deleted test cases entirely, e.g. where the corresponding issue said effectively "we'll live with the behavior" or "another repo is responsible for this now" (paraphrasing), or where the tests were initially added to the repo years ago already disabled and against bug numbers from some old bug database.
- In some cases, I replaced the link with the correct one.
- In some cases, I re-opened closed issues.

stephentoub commented:
@safern is investigating the enterprise-linux failure, which is happening in other PRs, too.
